homeassistant/config_entries.py

```diff
@@ -859,10 +883,14 @@ def _abort_if_unique_id_configured(
 
         for entry in self._async_current_entries():
             if entry.unique_id == self.unique_id:
-                if updates is not None and not updates.items() <= entry.data.items():
-                    self.hass.config_entries.async_update_entry(
+                if updates is not None:
+                    changed = self.hass.config_entries.async_update_entry(
                         entry, data={**entry.data, **updates}
```

Comment:
I'm not entirely sure why the whole changed data stuff if moved the async_update_entry in this case?

It adds a lot of code, while it already did prevent updating when there was no data changed.

Reply:
The goal was to make the check more robust since we previously only looked at `entry.data`.  Since the same problem is going to be there for `entry.options`, etc it seemed like it made sense to knock them all out.

Reply:
Right! My mind was on the config flow only case. Makes sense!